### PR TITLE
Revert "Revert "Jump search""

### PR DESCRIPTION
**Pull request by sunshinesway**
Reverts commonCereal/DDSA#19

Data Structure / Algorithm

    Jump search algorithm

Solution / Explanation of Implementation.

    Searches a list for a target value using the jump search method
    This algorithm assumes that the list is in ascending order
    Algorithm searches by jumping items in the list until items are greater than the target value
    Then the list is searched linearly beginning at the previous jump point until the target value is found
    -1 is returned if target value is not in the list, returns index value of the target value if present in the list

Semver Check

[ ] Major - Backwards Incompatible change on the API (includes removals)

[X] Minor - Adds or deprecates part of the API.

[ ] Patch - Doesn't effect API

Tag Reviewer(s)

@commonCereal

### CODE REVIEW

**Comment by sunshinesway** [comment]
Data Structure / Algorithm

    Jump search algorithm

Solution / Explanation of Implementation.

    Searches a list for a target value using the jump search method
    This algorithm assumes that the list is in ascending order
    Algorithm searches by jumping items in the list until items are greater than the target value
    Then the list is searched linearly beginning at the previous jump point until the target value is found
    -1 is returned if target value is not in the list, returns index value of the target value if present in the list

Semver Check

Major - Backwards Incompatible change on the API (includes removals)

Minor - Adds or deprecates part of the API.

    Patch - Doesn't effect API

Tag Reviewer(s)

@commonCereal

**Comment by commonCereal** [comment]
The [original PR](https://github.com/commonCereal/DDSA/pull/16) for this work was accidentally merged without proper approvals. Due to this, I had to revert this PR with a [Revert PR](https://github.com/commonCereal/DDSA/pull/19). This PR reverts the revert to allow @sunshinesway to get a proper approval on her work and get Hacktoberfest credit.

Note: After this I have updated the master branch protection rules to require at least one approval from maintainer to prevent this from happening to contributors who are new to Github.

**Review by commonCereal — APPROVED** [review]

**Pull request merged**
